src/Stratis.Bitcoin/Utilities/Guard.cs

```diff
@@ -63,5 +82,38 @@ public static string NotEmpty(string value, string parameterName)
 
             return value;
         }
+
+        /// <summary>
+        /// Checks a <see cref="string"/> is not null or empty.
+        /// </summary>
+        /// <param name="value">The string to check.</param>
+        public static void NotEmpty(string value)
+        {
+            if (ReferenceEquals(value, null))
+            {
+                string name = GetOriginalVariableName(value);
+                throw new ArgumentNullException(name);
+            }
+
+            if (value.Trim().Length == 0)
+            {
+                string name = GetOriginalVariableName(value);
+                throw new ArgumentException($"The string parameter {name} cannot be empty.");
+            }
+        }
+
+        /// <summary>
+        /// Gets the name of the original variable.
+        /// </summary>
+        /// <param name="obj">The variable to get the name of.</param>
+        private static string GetOriginalVariableName(object obj)
```

Comment:
This appears to be cumbersome work just to get the variable name when ArgumentNullException with a magic string is quite well handled by the IDE and even suggests you change it to nameof

![screenshot 2018-05-24 21 03 02](https://user-images.githubusercontent.com/444711/40510978-3c9989b6-5f96-11e8-879f-9e634077419a.png)

Especially as this Guard method is called a lot in the code, is this too much and unnecessary just to get a slightly better looking Guard at the top of the methods.

Or are there some other advantages that make it worth it ?

Reply:
>and even suggests you change it to nameof

what? I'm not sure what you've meant by that example.

>is this too much and unnecessary just to get a slightly better looking Guard at the top of the methods.

Yep, all of that to get better looking Guard. 
Also what do you mean by "too much"? No additional CPU is being consumed.

Reply:
Yes, I made a mistake on that one, I saw the method that got the variable name and made the assumption it was called every time. Apologies.

So yes, I do agree with it looking cleaner at the top of the method and therefore worth it.